### PR TITLE
fixed path with spaces in llvm-config

### LINKED ISSUE #2853: Compiling zig in folder with spaces fails
So I'm compiling zig inside my Google Drive folder (contains spaces) and it messes up some of the llvm library names. The error is: `LINK : fatal error LNK1181: cannot open input file 'C:\Users\ian\Google.obj'`

I found out the culprit is `llvm-config --libfiles` (inside Findllvm.cmake, line 70), which separates paths by spaces alone, making it ambiguous. There's an llvm bug report, but it hasn't been fixed in three years! https://bugs.llvm.org/show_bug.cgi?id=27743

This other project fixes it by concatting `llvm-config --libpath` with all `llvm-config --libnames`:
https://github.com/simit-lang/simit/issues/30
https://github.com/gkanwar/simit-1/commit/4457e58cec9d209e25ed5c5c5173ce6de049a1fc

The main thing I want to clear up before fixing this is what to do with LLVM_LIBDIRS. llvm-config only has one libdir, but Findllvm.cmake assumes there's an array of libdirs, and so I'm not sure this concat approach is correct.

### PR DESCRIPTION
**Pull request by IanLilleyT**
Fixes #2853

### CODE REVIEW

File: cmake/Findllvm.cmake [hunk 1]
```diff
@@ -67,47 +67,53 @@ NEED_TARGET("XCore")
...
-      COMMAND ${LLVM_CONFIG_EXE} --system-libs
-      OUTPUT_VARIABLE LLVM_SYSTEM_LIBS_SPACES
-      OUTPUT_STRIP_TRAILING_WHITESPACE)
-  string(REPLACE " " ";" LLVM_SYSTEM_LIBS "${LLVM_SYSTEM_LIBS_SPACES}")
+    COMMAND ${LLVM_CONFIG_EXE} --libs
+    OUTPUT_VARIABLE LLVM_LIBNAMES_SPACES
+    OUTPUT_STRIP_TRAILING_WHITESPACE)
+    string(REPLACE " " ";" LLVM_LIBNAMES "${LLVM_LIBNAMES_SPACES}")
+  foreach(LLVM_LIBNAME ${LLVM_LIBNAMES})
+    list(APPEND LLVM_LIBRARIES ${LLVM_LIBDIR}/${LLVM_LIBNAME})
```
Comment (andrewrk) [inline cmake/Findllvm.cmake]:
```suggestion
    list(APPEND LLVM_LIBRARIES "${LLVM_LIBDIR}/${LLVM_LIBNAME}")
```

in cmake you have to quote the shit out of everything

File: cmake/Findllvm.cmake [hunk 2]
```diff
@@ -67,47 +67,53 @@ NEED_TARGET("XCore")
...
 if(NOT LLVM_LIBRARIES)
   execute_process(
-      COMMAND ${LLVM_CONFIG_EXE} --libs
-      OUTPUT_VARIABLE LLVM_LIBRARIES_SPACES
-      OUTPUT_STRIP_TRAILING_WHITESPACE)
-  string(REPLACE " " ";" LLVM_LIBRARIES "${LLVM_LIBRARIES_SPACES}")
+    COMMAND ${LLVM_CONFIG_EXE} --libdir
+    OUTPUT_VARIABLE LLVM_LIBDIR_ORIG
+    OUTPUT_STRIP_TRAILING_WHITESPACE)
+  file(TO_CMAKE_PATH ${LLVM_LIBDIR_ORIG} LLVM_LIBDIR)
```
Comment (andrewrk) [inline cmake/Findllvm.cmake]:
```suggestion
  file(TO_CMAKE_PATH "${LLVM_LIBDIR_ORIG}" LLVM_LIBDIR)
```

File: cmake/Findllvm.cmake [hunk 3]
```diff
@@ -67,47 +67,53 @@ NEED_TARGET("XCore")
...
-      COMMAND ${LLVM_CONFIG_EXE} --system-libs --link-static
-      OUTPUT_VARIABLE LLVM_SYSTEM_LIBS_SPACES
-      OUTPUT_STRIP_TRAILING_WHITESPACE)
-  string(REPLACE " " ";" LLVM_SYSTEM_LIBS "${LLVM_SYSTEM_LIBS_SPACES}")
+    COMMAND ${LLVM_CONFIG_EXE} --libnames --link-static
+    OUTPUT_VARIABLE LLVM_LIBNAMES_SPACES
+    OUTPUT_STRIP_TRAILING_WHITESPACE)
+  string(REPLACE " " ";" LLVM_LIBNAMES "${LLVM_LIBNAMES_SPACES}")
+  foreach(LLVM_LIBNAME ${LLVM_LIBNAMES})
+    list(APPEND LLVM_LIBRARIES ${LLVM_LIBDIR}/${LLVM_LIBNAME})
```
Comment (andrewrk) [inline cmake/Findllvm.cmake]:
```suggestion
    list(APPEND LLVM_LIBRARIES "${LLVM_LIBDIR}/${LLVM_LIBNAME}")
```

File: cmake/Findllvm.cmake [hunk 4]
```diff
@@ -67,47 +67,53 @@ NEED_TARGET("XCore")
...
 if(NOT(CMAKE_BUILD_TYPE STREQUAL "Debug") OR ZIG_STATIC)
   execute_process(
-      COMMAND ${LLVM_CONFIG_EXE} --libfiles --link-static
-      OUTPUT_VARIABLE LLVM_LIBRARIES_SPACES
-      OUTPUT_STRIP_TRAILING_WHITESPACE)
-  string(REPLACE " " ";" LLVM_LIBRARIES "${LLVM_LIBRARIES_SPACES}")
+    COMMAND ${LLVM_CONFIG_EXE} --libdir --link-static
+    OUTPUT_VARIABLE LLVM_LIBDIR_ORIG
+    OUTPUT_STRIP_TRAILING_WHITESPACE)
+  file(TO_CMAKE_PATH ${LLVM_LIBDIR_ORIG} LLVM_LIBDIR)
```
Comment (andrewrk) [inline cmake/Findllvm.cmake]:
```suggestion
  file(TO_CMAKE_PATH "${LLVM_LIBDIR_ORIG}" LLVM_LIBDIR)
```

**Comment by IanLilleyT** [comment]
cool! thx for the review

**Comment by mikdusan** [comment]
```
rm -rf /Users/mike/project/zig/_build
mkdir -p /Users/mike/project/zig/_build
cmake -G Ninja /Users/mike/project/zig/master -DCMAKE_BUILD_TYPE=Debug -DCMAKE_INSTALL_PREFIX=/opt/zig -DCMAKE_PREFIX_PATH=/opt/llvm-8.0.1
-- The C compiler identification is AppleClang 11.0.0.11000028
-- The CXX compiler identification is AppleClang 11.0.0.11000028
```

...

```
-- Generating done
-- Build files have been written to: /Users/mike/project/zig/_build
ninja -j6
ninja: error: '/opt/llvm-8.0.1/lib/-lLLVMSymbolize', needed by 'zig0', missing and no known rule to make it
../zbuild:42:in `execute': error: command exited with error status: pid 20057 exit 1 (RuntimeError)
	from ../zbuild:88:in `<main>'
```

**Comment by Tetralux** [comment]
Breaks on linux x64 with:
```
Scanning dependancies of target zig0
[ 98%] Building CXX object CMakeFiles/zig0.dir/src/main.cpp.o
make[2]: *** No rule to make target '/usr/lib/-lLLVM-8', needed by 'zig0'. Stop.
make[1]: *** [CMakeFiles/Makefile2:349: CMakeFiles/zig0.dir/all] Error 2
make: *** [Makefile:130: all] Error 2
```

**Pull request merged**